csrc/host_ir/lower_to_communication.h

```diff
@@ -28,12 +33,10 @@ struct CommunicationInfo {
 // Composite expressions that are communication + compute are not supported.
 bool isCommunicationLayoutCompliant(Expr* expr);
 
-// Returns the communication info for the
-// (All)Gather/Scatter/ReduceScatter/(All)Reduce communication that may require
-// copying the input/output and reordering the allocation domain.
-// We assume that the expr has been decomposed and represented a single
-// communication. If multiple communications are present, this function will
-// raise an error.
+// Given an Expr that's known to be a communication, returns the communication
+// info: type and sharded IDs. We assume that the expr has been decomposed and
+// represented a single communication. If multiple communications are present or
+// 2D sharding, this function will raise an error.
```

Comment:
How will 2D sharding raise an error?

Reply:
```
in: iDIDx{m}, iDIDy{n}
out = set(in): iDIDx{m}, i{n}
```
When looking at DIDx, the code `create_communication_info`s `SendRecv`. When further looking at DIDy, the code attempts to `create_communication_info`s `Allgather` and fail at line 319.

Reply:
Got it.
In that case, we should not be creating `SendRecv` unconditionally. See my comment above.